src/headless/converse-chatboxes.js

```diff
@@ -964,6 +966,7 @@ converse.plugins.add('converse-chatboxes', {
                     'is_archived': this.isArchived(original_stanza),
                     'is_delayed': !!delay,
                     'is_spoiler': !!spoiler,
+                    'is_replace': !!replaced_id,
```

Comment:
`is_replace` is grammatically incorrect and I think it's less helpful to store a boolean than storing the actual ID of the message being replaced.

So instead of adding `is_replace`, use `replaced_id` directly.

Then in `correctMessage` on line 555 please change

    const message = this.messages.findWhere({'msgid': attrs.msgid, 'from': attrs.from});

to 

     const message = this.messages.findWhere({'msgid': attrs.replaced_id, 'from': attrs.from});

I'm not sure why I didn't initially do it like this, it now seems so obvious. Hopefully I'm not missing some important detail.

Reply:
With this new logic, does it still make sense to set the replaced_id as msgid? (`const msgid = replaced_id || stanza.getAttribute('id') || original_stanza.getAttribute('id');`)

Reply:
I think so yes. We at this point don't yet know whether we have the message being replaced and `replaced_id` points to that messsage's ID. If such a message doesn't exist, we use the data from this message to create it.

Reply:
In fact, all corrections and the original corrected message are all collapsed into one Message object, but the point still stands.

Reply:
OK done, and verified that message corrections still work.